docs/docs/vm-fundamentals/hints.md

```diff
@@ -2,6 +2,236 @@
 sidebar_position: 3
 ---
 
-# Hints
+# Hint
+  ## HINTS  IN CAIRO VIRTUAL MACHINE
+Hints in Cairo are pieces of Python code only seen and executed by the prover and are not included in the bytecode. They instruct the prover on how to handle nondeterministic instructions. These nondeterministic instructions are programs that have different outcomes at execution.   
+Hints are guides that developers can provide to the Cairo virtual machine to optimize execution or provide additional information about the program. These hints help improve performance or help the VM make better decisions during the execution of programs. 
+
+## TYPES OF HINTS IN CARIO VIRTUAL MACHINE
+The hints in Cairo Virtual Machine that influence the behavior of programs include:
+### EXECUTION HINTS
+ they guide the execution of cairo programs. They include guides on loop unrolling, parallel execution, or optimizations to enhance. It influences how the VM schedules and executes instructions, thereby improving performance.
+In Cairo, execution hints use the *hint* keyword, followed by a *hint name* and *a value*.
+
+ @example:
+
+ ```cairo  
+      func main() -> felt*:
+      // some code here
+      hint memory_behavior = MemoryBehavior.Sequential;
+      // more code here
+  ```
+  The memory_behavior hint informs the Cairo compiler that the code following the hint   
+ sequentially accesses memory.
+
+ @example:
+
+ ``` cairo
+     func main() {
+         // example of loop unrolling hint
+         @loop_unroll(4)
+         for i = 0 to 10 {
+         // Loop body
+        // example of parallel execution hint
+         }
+         @parallel_execution
+         {
+         // Code block for execution in parallel
+         }
+     }
+ ```
+
+### MEMORY HINTS
+ These hints are memory aids that provide additional information to the prover on computing values in the Cairo program. They are functional when provers can't handle certain expressions implicitly.
```

Comment:
That is not actual essence of hints. Cairo is a non-deterministic language, meaning, imagine a Turing Machine where you can visit different states at the same time (in theory). In practice, that's super expensive and can grow exponentially in memory. Hints are used to avoid all of this, and "hint" the VM to which state it should jump into (of the many possible)